### PR TITLE
[renderer] js version.

### PR DESCRIPTION
**Pull request by makasim**
I created js version of [php renderer class](https://github.com/arnaud-lb/fselectmenu/blob/master/lib/FSelectMenu/Renderer.php). The main goal was to keep all interfaces and behaviors as is in php. So I used some functions from phpjs.org like `array_replace_recursive`.

Usage:

``` javascript
    require([
        'fselectmenu/fselectmenu', 
        'fselectmenu/renderer'], 
        function(FSelectMenu, FSelectMenuRenderer) {
        var value = "foo";
        var choices = {
            "foo": "FooTitle",
            "bar": "BarTitle",
            "ololo": {
                "val1": "Val1Title",
                "val2": "Val2Title"
            }
        };
        var options = {};

        FSelectMenu.init();

        $('#test-fselectmenu-container').html(FSelectMenuRenderer.render(value, choices, options));
    });
```

### CODE REVIEW

**Comment by makasim** [comment]
the js version has only one difference: it does not support translation.

**Comment by makasim** [comment]
@arnaud-lb I moved your provider data to yaml file. It allows to create same tests for js renderer. Give me your feedback about it.

File: test/FSelectMenu/Tests/render-test-data.yaml [hunk 1]
```diff
@@ -0,0 +1,527 @@
...
+6 - Custom class:
+    value: ''
+    choices: {  }
+    options: { attrs: { class: fselectmenu-style-foo } }
+    output: | 
+        <span class="fselectmenu-style-foo fselectmenu fselectmenu-events fselectmenu-value-" tabindex="0">
+        <select class=" fselectmenu-native">
+        </select>
+        <span class="fselectmenu-label-wrapper">
+        <span class="fselectmenu-label">&#xA0;</span>
```
Comment (makasim) [inline test/FSelectMenu/Tests/render-test-data.yaml]: @arnaud-lb while moving the data to yaml I found phpunit complain to this string `<span class="fselectmenu-label"> </span>` the diff was actually in: `space` -> `&#xA0;` . Could you check whether it is what it should be.

**Comment by arnaud-lb** [comment]
> I moved your provider data to yaml file. It allows to create same tests for js renderer. Give me your feedback about it.

Awesome :)

File: test/FSelectMenu/Tests/render-test-data.yaml (hunk 1 above)
Comment (arnaud-lb) [inline test/FSelectMenu/Tests/render-test-data.yaml]: Yes, empty labels are rendered as non-break-space to force rendering of the label in the browser

**Comment by makasim** [comment]
@arnaud-lb add js test. The tests passed without js errors. but the rendered output is not equals what expected. I will try to fix it later when I have time

**Comment by makasim** [comment]
the php render set an empty  value as `\xC2\xA0` and it is rendered to `<span class="fselectmenu-label">&#xA0;</span>`

the js version has the same code but it is rendered to `<span class="fselectmenu-label">&#xC2;&#xA0;</span>`

any thoughts?

**Comment by arnaud-lb** [comment]
this may be an encoding issue; try prepending `<?xml version="1.0" encoding="utf-8"?>` to strings before passing them to assertXmlStringEqualsXmlString

**Comment by makasim** [comment]
no it does not help:

```
-    <span class="fselectmenu-label"> </span>
+    <span class="fselectmenu-label">Â </span>
```

I can reach the same result if I set `\xC2\xA0` and `\xA0"` in js renderer.

**Comment by arnaud-lb** [comment]
It works here with the following changes:

``` diff
--- a/test/FSelectMenu/Tests/RendererTest.php
+++ b/test/FSelectMenu/Tests/RendererTest.php
@@ -18,7 +18,10 @@ class RendererTest extends \PHPUnit_Framework_TestCase

         $result = str_replace('><', ">\n<", $result);

-        $this->assertXmlStringEqualsXmlString($expect, $result);
+        $this->assertXmlStringEqualsXmlString(
+            '<?xml version="1.0" encoding="utf-8"?>'.$expect
+            , '<?xml version="1.0" encoding="utf-8"?>'.$result
+        );
     }

     public function provideRenderTestData()

--- a/test/FSelectMenu/Tests/render-test-data.yaml
+++ b/test/FSelectMenu/Tests/render-test-data.yaml
@@ -7,7 +7,7 @@
         <select class=" fselectmenu-native">
         </select>
         <span class="fselectmenu-label-wrapper">
-        <span class="fselectmenu-label">&#xA0;</span>
+        <span class="fselectmenu-label">&#x00A0;</span>
         <span class="fselectmenu-icon">
         </span>
         </span>
```

**Comment by makasim** [comment]
not work for me

**Comment by makasim** [comment]
Tests execution:

```
.............F....F.......................
```

**Comment by makasim** [comment]
One faild test - because translation (it does not work in js) and the second one is a bug with prefered choices.

**Comment by makasim** [comment]
@arnaud-lb @jeremymarc done

**Pull request merged**